server/server.go

```diff
@@ -272,11 +276,13 @@ func (s *Server) serviceAccountIdentityHandler(logger *log.Entry, w http.Respons
 		return
 	}
 
+	w.Header().Set(defaultFlavorHeaderName, defaultFlavorHeaderValue)
```

Comment:
If we end up in the above branches (`if audience == ""` or `err != nil`) we won't return any header, Google does return it:
```
curl -v http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity >/dev/null -H "Metadata-Flavor: Google"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 169.254.169.254...
* TCP_NODELAY set
* Connected to metadata.google.internal (169.254.169.254) port 80 (#0)
> GET /computeMetadata/v1/instance/service-accounts/default/identity HTTP/1.1
> Host: metadata.google.internal
> User-Agent: curl/7.52.1
> Accept: */*
> Metadata-Flavor: Google
>
< HTTP/1.1 400 Bad Request
< Metadata-Flavor: Google
< Date: Tue, 09 Jul 2019 17:52:48 GMT
< Content-Type: text/html
< Server: Metadata Server for VM
< Content-Length: 37
< X-XSS-Protection: 0
< X-Frame-Options: SAMEORIGIN
<
{ [37 bytes data]
* Curl_http_done: called premature == 0
100    37  100    37    0     0   2740      0 --:--:-- --:--:-- --:--:--  2846
* Connection #0 to host metadata.google.internal left intact
```